### ISSUE TITLE
There are still overlapping peaks in the produced consensus peak set

### ISSUE THREAD

**Issue by yqlan** [databio/pepatac#253]
I first ran `looper run` without a reference peak set, then ran `looper runp`.
It seems there are still overlapping peaks in the produced consensus peak set, is this right?

**Comment by jpsmith5** [comment]
Yes, that's expected. Both the within-sample `_peaks_normalized.narrowPeak` and the project-level consensus peak set use a "keep highest-scoring peak per overlap cluster" heuristic (`PEPATACr::reducePeaks` and `PEPATACr::collapsePeaks`), which deduplicates MACS2's source peaks but doesn't guarantee strictly disjoint ranges in all configurations — e.g., if A overlaps B and B overlaps C but A and C don't overlap each other, the heuristic can keep both A and C. If you need fully non-overlapping intervals downstream, post-process with `bedtools merge` or `GenomicRanges::reduce()`. The consensus-peak method is being revisited in PR #211. Closing as answered.
